pyrefly/lib/error/kind.rs

```diff
@@ -189,6 +189,19 @@ pub enum ErrorKind {
     UnsupportedOperand,
 }
 
+impl std::str::FromStr for ErrorKind {
+    type Err = ();
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        ERROR_KIND_CACHE
+            .iter()
+            .zip(enum_iterator::all::<ErrorKind>())
```

Comment:
instead of zipping here, could we make the `ERROR_KIND_CACHE` a `SmallMap<String, ErrorKind>`?

The reason we have a cache for this is because `enum_iterator` is apparently pretty slow, so we only want to do this once.